client/lib/posts/utils.js

```diff
@@ -49,6 +49,12 @@ var utils = {
 			if ( site.options.is_mapped_domain ) {
 				previewUrl = previewUrl.replace( site.URL, site.options.unmapped_url );
 			}
+			if ( site.options.frame_nonce ) {
+				parsed = url.parse( previewUrl, true );
+				parsed.query['frame-nonce'] = site.options.frame_nonce;
+				delete parsed.search;
+				previewUrl = url.format( parsed );
+			}
```

Comment:
As much as it pains me to say, I'm gonna punt on that, since fixing everything here has already turned into a giant interdimensional rabbit hole.